src/historic-quote/historic-quote.service.ts

```diff
@@ -17,13 +17,35 @@ type Candlestick = {
   close: string;
   high: string;
   low: string;
+  provider: string;
+};
+
+type PriceProvider = 'coinmarketcap' | 'codex';
+
+interface ProviderConfig {
+  name: PriceProvider;
+  enabled: boolean;
+  priority: number;
+}
+
+type BlockchainProviderConfig = {
+  [key in BlockchainType]: ProviderConfig[];
 };
 
 @Injectable()
 export class HistoricQuoteService implements OnModuleInit {
   private isPolling = false;
   private readonly intervalDuration: number;
   private shouldPollQuotes: boolean;
+  private priceProviders: BlockchainProviderConfig = {
+    [BlockchainType.Ethereum]: [
+      { name: 'coinmarketcap', enabled: true, priority: 1 },
+      { name: 'codex', enabled: true, priority: 2 },
+    ],
+    [BlockchainType.Sei]: [{ name: 'codex', enabled: true, priority: 1 }],
```

Comment:
wouldn't it be easier to pass them sorted by priority, saving you the need to maintain the number and avoid accidentally repeating it, plus no need to sort later on?

Reply:
done